### PR TITLE
Add e2e test for sources with targetpath

### PR DESCRIPTION
**Pull request by shashwathi**
This PR includes e2e test of using `targetPath` in a valid use-case i.e., source copied into targetpath and running go tests.

### CODE REVIEW

File: test/sources/test-gopath.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: build.knative.dev/v1alpha1
+kind: Build
+metadata:
+  name: test-gopath-test
+  labels:
+    expect: failed
```
Comment (ImJasonH) [inline test/sources/test-gopath.yaml]: Should this be `expect: succeeded` ?

File: test/sources/test-gopath.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+spec:
+  sources:
+  - name: build-git
+    targetPath: go/src/github.com/knative/build
+    git:
+      url: https://github.com/knative/build
+      revision: master
+  steps:
+  - name: test-serving-build
```
Comment (ImJasonH) [inline test/sources/test-gopath.yaml]: `test-build` ?

File: test/sources/test-gopath.yaml (hunk 1 above)
Comment (shashwathi) [reply test/sources/test-gopath.yaml]: Yes. How did this pass locally is a mystery I will never understand.!

**Comment by adrcunha** [comment]
/hold

Just in case Jason has any other comment.

**Pull request merged**